### PR TITLE
IRIS-4724 | Fix tests for deleting all posts by user

### PR DESCRIPTION
**Pull request by tortila**
[Ticket](https://wikia-inc.atlassian.net/browse/IRIS-4724)

At this point, where mobile and desktop functionality are that different, I think it's reasonable to refactor the tests to use 2 different class for mobile and desktop. Here is [a ticket](https://wikia-inc.atlassian.net/browse/IRIS-4732) to follow up.

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 1]
```diff
@@ -193,43 +197,51 @@ public void modUserDesktopDeleteAllOptionNotVisibleOnDifferentWiki() {
...
 
   // HELPER METHODS
 
-  private boolean deleteAllOptionNotVisible(String userId) {
-    return getDeleteAllButton(userId).isNotVisible();
+  private boolean deleteAllOptionNotVisibleMobile(String userId) {
+    return getDeleteAllButtonMobile(userId).isNotVisible();
+  }
+
+  private boolean deleteAllOptionNotVisibleDesktop(String userId) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MAJOR Remove this unused private "deleteAllOptionNotVisibleDesktop" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java [hunk 2]
```diff
@@ -193,43 +197,51 @@ public void modUserDesktopDeleteAllOptionNotVisibleOnDifferentWiki() {
...
   public void modUserMobileCanDeleteAllPostsOnModdedWiki() {
-    assertTrue(deleteAllAndReturnFirst().isDeleted());
+    assertTrue(deleteAllAndReturnFirstMobile().isDeleted());
   }
 
   // HELPER METHODS
 
-  private boolean deleteAllOptionNotVisible(String userId) {
-    return getDeleteAllButton(userId).isNotVisible();
+  private boolean deleteAllOptionNotVisibleMobile(String userId) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java]: MAJOR Remove this unused private "deleteAllOptionNotVisibleMobile" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DeleteAllPostsByUserTests.java#L41](https://github.com/Wikia/selenium-tests/blob/1b806a7a991efe8909f852813b56e63864fe6bc0/src/test/java/com/wikia/webdriver/testcases/discussions/DeleteAllPostsByUserTests.java#L41): Remove this unused private "setUp" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/UserPostsPage.java [hunk 3]
```diff
@@ -20,6 +24,12 @@
 
   @Getter(lazy = true) private final DeleteAllButton deleteAll = new DeleteAllButton();
 
+  /**
+   * moderation section visible to mod+ users in mobile view
```
Comment (slayful) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/UserPostsPage.java]: `Moderation section visible to mod+ users in mobile view.`?

**Pull request merged**